frontend/packages/ceph-storage-plugin/src/components/bucket-class/wizard-pages/namespace-store-pages/multi-namespace-store.tsx

```diff
@@ -23,22 +23,26 @@ export const MultiNamespaceStorePage: React.FC<MultiNamespaceStoreProps> = React
     const openModal = () => NamespaceStoreModal({ namespace });
     const [selectedCount, setSelectedCount] = React.useState(state.readNamespaceStore.length);
     const [enabledItems, setEnabledItems] = React.useState([]);
+    const [selectedItem, setSelectedItem] = React.useState('');
 
     React.useEffect(() => {
       const readItems = state.readNamespaceStore.map(getName);
       setEnabledItems(readItems);
-    }, [state.readNamespaceStore]);
+      setSelectedItem(getName(state.writeNamespaceStore[0]) || '');
+    }, [state.readNamespaceStore, state.writeNamespaceStore, selectedItem]);
 
     const onSelectNamespaceStoreTable = (selectedNamespaceStore: NamespaceStoreKind[]) => {
       dispatch({ type: 'setReadNamespaceStore', value: selectedNamespaceStore });
       setSelectedCount(selectedNamespaceStore.length);
       if (!selectedNamespaceStore.map(getName).includes(getName(state.writeNamespaceStore[0]))) {
         dispatch({ type: 'setWriteNamespaceStore', value: [] });
+        setSelectedItem('');
       }
     };
 
     const onSelectNamespaceStoreDropdown = (selectedNamespaceStore: NamespaceStoreKind) => {
       dispatch({ type: 'setWriteNamespaceStore', value: [selectedNamespaceStore] });
+      setSelectedItem(getName(state.writeNamespaceStore[0]));
```

Comment:
dispatch is an async update so state.writeNamespaceStore[0] wont gives you the latest value

try `setSelectedItem(getName(selectedNamespaceStore));`

Reply:
makes sense ! thanks